File: py/ipmailer.py
import urllib2, json, smtplib, os
from urllib2 import URLError

email = "example@gmail.com"
password = "123456"
mailServer = "smtp.example.com"
mailPort = 587
to = ['youremail@yourprovider.com']
subject = "IP Changed"

def checkIp():
	try:
		jsonip = json.loads(urllib2.urlopen("http://jsonip.com/").read())
		f = open('mylastip', 'a+')
		f.seek(0)
		lastIp = f.read()
		if lastIp != jsonip['ip'] :
			sendMail("Hello, my ip Changed to: " + jsonip['ip'])
			if os.popen('uname').read() == "Darwin\n" :
			    sendNotification("IPMailer", "IP changed to " + jsonip['ip'])
			f.seek(0)
			f.truncate()
			f.write(jsonip['ip'])
		f.close()
	except URLError, e:
		return
		
def sendNotification(t, m):
    os.system('./terminal-notifier -title "'+t+'" -message "'+m+'"' )

def sendMail(message):
	try:
	    server = smtplib.SMTP(mailServer, mailPort)
	    server.ehlo()
	    server.starttls()
	    server.login(email, password)
	    server.sendmail(email, to, message)
	    server.close()
	    print 'successfully sent the mail'
	except:
	    print "failed to send mail"

checkIp()

File: php/ipmailer.php
#!/usr/bin/php
<?php
#About: this script sends an email if your IP address change, it's supposed to run on cron

$name= "Your name";
$from="Your email";
$password="123456";
$mailServer="ssl://smtp.googlemail.com";
$mailPort=465;
$to="Your rcpt";
$subject="IP Changed";

function sendMail($message){
    global $name, $from, $password, $mailServer, $mailPort, $to, $subject;
    require 'PHPMailer/PHPMailerAutoload.php';

    $email = new PHPMailer();
    $email->isSMTP();
    $email->SMTPDebug = false;
    $email->Host = $mailServer;
    $email->SMTPAuth = true;
    $email->Port     = $mailPort;
    $email->Username = $from;
    $email->Password = $password;
    $email->FromName = $name;
    $email->From     = $from;
    $email->AddAddress($to, $to);
    $email->Subject  = $subject;
    $email->Body = $message;

    if(!$email->send())
        echo 'Mailer Error: ' . $email->ErrorInfo;
}
function checkIp(){
    $json = json_decode(file_get_contents('http://jsonip.com'));
    $jsonip = $json->ip;
    $f=fopen("lastIp",'a+b');
    $lastIp=stream_get_contents($f);
    if ($jsonip != $lastIp){
        sendMail("Hello, my ip changed to: ".$jsonip);
        ftruncate($f,0);
        fwrite($f,$jsonip);
    }
    fclose($f);
}
checkIp();

?>